interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -51,27 +52,70 @@ type Prompt struct {
 	Timestamp    time.Time          `json:"timestamp"`
 	Snap         string             `json:"snap"`
 	Interface    string             `json:"interface"`
-	Constraints  *PromptConstraints `json:"constraints"`
+	Constraints  *promptConstraints `json:"constraints"`
```

Comment:
I'm not sure this is less confusing. What to do consumer of this package do with Prompt structs?

Reply:
one suggestion would be to actually go back to the simpler state of things and strip the json tags in the package here and add them back later or do something else when we see how they are used

Reply:
The consumer is the daemon (via the `apparmorprompting` ifacestate manager), which marshals the `Prompt` and sends it over the API when requested. If/when we persist prompts across snapd restarts, we'll need to marshal and unmarshal prompts to disk, but for now we just store in memory and send as json over the API.

Reply:
Until we actually need to marshal prompts to send over the API (or store on disk), we'll remove the json annotations, struct, and marshal function.